### ISSUE TITLE
Docs don't clarify that `stack` can be used as a `cabal-install` replacement

### ISSUE THREAD

**Issue by osfameron** [commercialhaskell/stack#1816]
I was just recommended to use `stack` instead of `cabal-install` in #haskell.

This confused me, as I'd looked at http://docs.haskellstack.org/en/stable/README/ earlier today and thought "that looks interesting, I must come back to that", but had no idea that it could be used to install modules.  (e.g. I'd assumed it was for setting up a project environment, somewhat in the style of perl brew/ruby Bundler/python virtualenv etc.)

The only reference to `cabal` on the front-page is as a means to install `stack` itself.  (+ a few tangential references to the `.cabal` file.)

It might be worth clarifying with some wording along the lines of "Stack may be used to install Haskell modules (and may be used as a replacement for cabal-install)" ?

**Comment by harendra-kumar** [comment]
I agree that's something that will help the newcomers understand the scope of the tool. It was not clear to me as well when I first looked at stack. I figured that out only after installing and using it. 

@osfameron will it help if we add the following to the existing bullet items on the README page?:
```
* Installing packages from Stackage and Hackage
```

**Comment by borsboom** [comment]
It's a bit tricky to word, because in a sense Stack does _not_ replace cabal-install, since there are all sorts of things you can do with cabal-install that you can't do with Stack (such as install packages into GHC's global or user package databases).  Of course, this is deliberate as most of these are things you probably _shouldn't_ do, and many Stack users will never have any reason to use cabal-install.

Maybe more accurate to say that Stack can replace cabal-install _sandboxes_.

**Comment by mgsloan** [comment]
We could say that `stack` installs dependencies and builds your code, similarly to `cabal-install`.  So it to doesn't necessarily do the same sorts of things, but achieves the same goal.

**Comment by harendra-kumar** [comment]
That's covered by `Installing packages needed for your project.`. We can perhaps make it more explicit as you suggested above. What is not at all clear is that stack can also install packages from Stackage/Hackage in `.local/bin` and what `stack unpack/stack init/stack install` can do for example.

**Comment by osfameron** [comment]
Thanks - I think you're right that it's hard to find an accurate yet welcoming description for this!

It's true that "Installing packages needed for your project." does in fact mention that it can install modules (I wonder if I was distracted by the animated gif to the right -- I'd recommend removing it/replacing with a nicer link to an asciinema session?)

Perhaps it needs a "how does this fit into the existing toolchain?"  e.g. a few subheadings with "If you already use `cabal-install`|`stackage-cli`", that suggest whether you should switch to `stack`, or use both for different things?

**Comment by mgsloan** [comment]
Yeah, it'd also be good to cover using `stack exec` with things like `ghc-pkg` and `ghci`.  Folks might not realize that `stack exec` is providing GHC_PACKAGE_PATH, so those commands will use the correct package databases.

**Comment by sjakobi** [comment]
How about saying

> Installing Cabal packages needed for your project.

instead of

> Installing packages needed for your project.

? This way "cabal" is already mentioned and people might have an association to `cabal-install`.

OTOH some people may not know that "Cabal" is the Haskell package format…

Or we add a small section above ["How to install"]
(https://docs.haskellstack.org/en/stable/README/#how-to-install):

> #### So `stack` is like `cabal`?

> Kind of. `stack` and `cabal`/`cabal-install` have a large feature overlap but solve many things slightly differently.
>  If you want to program in Haskell, `stack` will likely provide for most of your immediate needs.

Or:

> `stack` is intended as an alternative to the `cabal`/`cabal-install` build tool.

**Comment by sjakobi** [comment]
Can someone help bikeshedding possible resolutions here?

**Comment by kva1966** [comment]
As a user of Haskell packages (e.g. `pandoc` and filters), rather than a Haskell programmer: I will say that while `stack` may not completely replace `cabal`, basic installs of packages are really welcome. 

The core idea of having per-project, reproducible builds is awesome, but I like that `stack` also has a so-called `global-project` as an abstraction: it allows me a way of using a more recent Haskell distribution and associated recent packages than those provided by the OS (talking to you Debian and LMDE!). 

Unfortunately, OS-provided `cabal`, as far as I can tell, does not (easily?) allow me to use a different, more recent GHC, so stack works well for this use.

My input then is to incorporate something along the lines of:

* Stack's main aim is to isolate dependencies for a project.
* However, Stack also has an implicit `global-project`, to install local user-wide dependencies for experimentation. These dependencies do NOT get inherited by any project created via `stack new`.
* Global project configuration is stored in `$HOME/<path-to-yaml>`

In short, I would like the abstraction of the global project to be exposed earlier on in the docs.

Btw, awesome work with the resolver providing recommended actions -- e.g. note on adding extra-deps.
